src/mon/MonClient.cc

```diff
@@ -220,35 +184,40 @@ int MonClient::ping_monitor(const string &mon_id, string *result_reply)
     return -EINVAL;
   } else if (!monmap.contains(new_mon_id)) {
     ldout(cct, 10) << __func__ << " no such monitor 'mon." << new_mon_id << "'"
-                   << dendl;
+		   << dendl;
     return -ENOENT;
   }
 
-  MonClientPinger *pinger = new MonClientPinger(cct, result_reply);
+  std::unique_ptr<MonClientPinger> pinger(
+    new MonClientPinger(cct, result_reply));
+
 
-  Messenger *smsgr = Messenger::create_client_messenger(cct, "temp_ping_client");
-  smsgr->add_dispatcher_head(pinger);
+  auto dm = [this](Messenger* m) {
+    m->shutdown();
```

Comment:
Is m guaranteed to be valid at this point? It's not the case in the code I see here, but is there any chance of the unique_ptr<> ever being default constructed and hitting this Deleter?